### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by karolina-rachuta**
Created a custom cartridge that extends the Home-Show endpoint and adds some JSON data to the view data, added a template to see the message

### CODE REVIEW

File: cartridges/app_custom_hw/cartridge/controllers/Home.js [hunk 1]
```diff
@@ -5,14 +5,19 @@
...
 
 server.extend(module.superModule);
 
 server.append('Show', function (req, res, next) {
     var viewData = res.getViewData();
     viewData = {
-        messege: 'there is a new cartridge KR'
-    };
+        message: 'there is a new cartridge KR'
+    }|
```
Comment (isergeev19) [inline cartridges/app_custom_hw/cartridge/controllers/Home.js]: There seems to be a typo

Comment (karolina-rachuta) [reply cartridges/app_custom_hw/cartridge/controllers/Home.js]: ok, I deleted it (commit "typo before if")

File: cartridges/app_storefront_base/cartridge/controllers/Home.js [hunk 2]
```diff
@@ -23,23 +24,27 @@ var pageMetaData = require('*/cartridge/scripts/middleware/pageMetaData');
  * @param {renders} - isml
  * @param {serverfunction} - get
  */
-server.get('Show', consentTracking.consent, cache.applyDefaultCache, function (req, res, next) {
+server.get('Show', userLoggedIn.validateLoggedIn, consentTracking.consent, cache.applyDefaultCache, function (req, res, next) {
```
[the remark this answers is not in the document]
Comment (karolina-rachuta) [reply cartridges/app_storefront_base/cartridge/controllers/Home.js:27]: I undone the changes in the base cartridge and added them to custom cartridge

[the remark this answers is not in the document]
Comment (karolina-rachuta) [reply cartridges/app_storefront_base/cartridge/controllers/Home.js]: Ok I fixed the hook problem, I needed to move the hook.json file and package.json file one level up from a cartridge directory, before I put them on inside cartridge directory

File: cartridges/app_custom_hw/cartridge/controllers/Home.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+var server = require('server');
+var userLoggedIn = require('*/cartridge/scripts/middleware/userLoggedIn');
+var HookMgr = require('dw/system/HookMgr');
+
+server.extend(module.superModule);
+
+server.prepend('Show', userLoggedIn.validateLoggedIn, function (req, res, next) {
+    var viewData = res.getViewData();
+    viewData = {
```
Comment (isergeev19) [inline cartridges/app_custom_hw/cartridge/controllers/Home.js:15]: In general, you should use setViewData, so you set only the new data, and don't overwrite existing viewdata

**Review by isergeev19 — APPROVED** [review]

**Pull request merged**